From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Chris Bobbe <cbobbe@zulip.com>
Date: Thu, 11 Apr 2024 16:22:37 -0700
Subject: [shortcut] rnav: Copy-paste translated Props type rather than
 importing

I don't really know what this @react-native-masked-view/masked-view
stuff is about. This @react-navigation/elements package lists it as
a *dev* dependency and calls it "optional":
  https://github.com/react-navigation/react-navigation/blob/6.x/packages/elements/src/MaskedViewNative.tsx#L17

Anyway, the type we want from it is simple, so we just translate it
and copy-paste it here, instead of adding the dependency and giving
more work to tools/tsflower.
---
 .../typescript/src/MaskedViewNative.js.flow    | 18 ++++++------------
 1 file changed, 6 insertions(+), 12 deletions(-)

diff --git types/@react-navigation/elements/lib/typescript/src/MaskedViewNative.js.flow types/@react-navigation/elements/lib/typescript/src/MaskedViewNative.js.flow
index 39da71d49..cc5fc2306 100644
--- types/@react-navigation/elements/lib/typescript/src/MaskedViewNative.js.flow
+++ types/@react-navigation/elements/lib/typescript/src/MaskedViewNative.js.flow
@@ -1,19 +1,13 @@
 /* @flow
  * @generated by TsFlower
  */
-import typeof * as $tsflower_import_typeof$_$_40_react_2d_native_2d_masked_2d_view_2f_masked_2d_view from '@react-native-masked-view/masked-view';
+import type { JSX$Element as $tsflower_subst$React$JSX$Element } from 'tsflower/subst/react';
+import type { ViewProps as $tsflower_subst$RN$ViewProps } from 'tsflower/subst/react-native';
 
-import type {
-  ComponentProps as $tsflower_subst$React$ComponentProps,
-  JSX$Element as $tsflower_subst$React$JSX$Element,
-} from 'tsflower/subst/react';
-
-import * as React from 'react';
-type MaskedViewType = $ElementType<
-  $tsflower_import_typeof$_$_40_react_2d_native_2d_masked_2d_view_2f_masked_2d_view,
-  'default',
->;
-type Props = $tsflower_subst$React$ComponentProps<MaskedViewType> & {
+type Props = {
+  ...$tsflower_subst$RN$ViewProps,
+  maskElement: React$Element<React$ElementType>,
+  androidRenderingMode?: 'software' | 'hardware',
   children: React$Element<React$ElementType>,
   ...
 };
-- 
2.32.0

